### PR TITLE
Add support for deployment with helm

### PR DESCRIPTION
**Pull request by yujunz**

### CODE REVIEW

File: kubernetes/clearwater/values.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Default values for clearwater.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+image:
+  path: quay.io/clearwater_ims
```
Comment (johadalin) [inline kubernetes/clearwater/values.yaml]: I believe this is currently pointing to your own build of Clearwater. Would be much better as a parameter that the config generation script can fill in.

File: kubernetes/clearwater/values.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  tag: autobuild
+hss:
```
Comment (johadalin) [inline kubernetes/clearwater/values.yaml]: I can't see this being used anywhere. I may be misunderstanding how helm works, as we have relatively little experience with it. Can you explain what purpose this entry in the values.yaml has?

File: kubernetes/clearwater/.helmignore [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Patterns to ignore when building packages.
```
Comment (johadalin) [inline kubernetes/clearwater/.helmignore]: Is this the same file taken from the example helm package? 
I don't believe we will need most of this, and probably don't need any of these entries. Do you agree?

If so i think we should just remove this file

File: README.md [hunk 4]
```diff
@@ -139,10 +139,22 @@ If the above requirements are not met (external devices cannot resolve Kubernete
...
+
 To deploy the images, you should simply run `kubectl apply -f clearwater-docker/kubernetes`.  It may take a minute or so before the deployment is fully established, the load balancer is created, and the deployment is ready to accept calls.
 
 Note this will deploy all containers.  If you don't need e.g. Bono, Homestead-prov, Ellis etc. then just move the corresponding svc and depl files out of the directory before running the create command.
 
+#### Using helm
+
+To create a helm chart, you should simply run `./k8s-genchart`.
+
+If you want to use your own docker registry, modify `clearwater/values.yaml`.
```
Comment (johadalin) [inline README.md]: If you're able to make the changes to build Helm config using the k8s-gencfg script, we won't need this :)
I'm assuming users will still have had to the up the config-map, as in the steps above? Haven't got helm installed on the rig i have available so haven't been able to test myself

**Review by johadalin — CHANGES_REQUESTED** [review]
Hi Yujunz,
Firstly, thanks a lot for sending this over! It's really great that you're getting involved in enhancing the project to use Helm. Should be able to work this out and get it in soon, and then we can see what else we can add.

I've put a few specific points in line, but i think this would be a lot nicer if we can merge the function you're adding in with the existing `k8s-gencfg` script, and moving things around to be a bit clearer. I think the right approach would be doing the following:

* creating a new `helm/` directory under the `kubernetes/` directory
* Moving the Chart.yaml and values.yaml to under `helm/`
* Parameterising the values.yaml file, similar to what is in our current templates
* Enhancing the `k8s-gencfg` script with a new function/logic that:
    * creates the `clearwater/` helm package directory
	* Moves the files in `helm/` across, filling in the parameterised image path and tag values (as we do for the existing templates)
	* Copies the main clearwater yaml files across to `clearwater/`
* add the whole `clearwater/` directory to the gitignore

This would mean users should only need to run the single config generation command, and we only have one source of truth for the image path and tag, not having values hardcoded in the values.yaml.

What do you think about this approach?

File: kubernetes/clearwater/values.yaml (hunk 1 above)
Comment (yujunz) [reply kubernetes/clearwater/values.yaml]: Agree

File: kubernetes/clearwater/values.yaml (hunk 2 above)
Comment (yujunz) [reply kubernetes/clearwater/values.yaml]: It was used for creating config map. It seems that I forgot to add the template.

File: kubernetes/clearwater/.helmignore (hunk 3 above)
Comment (yujunz) [reply kubernetes/clearwater/.helmignore]: Agree

File: README.md (hunk 4 above)
Comment (yujunz) [reply README.md]: We may include the ConfigMap with default values in helm chart. It is quite convenient to override the default values from command arguments.

**Comment by yujunz** [comment]
+1 for a single command and one source of truth. I will update the patch according to your comments :-)

**Comment by yujunz** [comment]
Ready for review @johadalin 

Tested in GKE.

**Comment by johadalin** [comment]
@yujunz this looks great. All fits nicely into the `gencfg` script now.
Tested it out on our kubernetes rig as well, and it all seems to work as expected, at least for basic function. Hope to have some time soon to play around with the wider function that Helm might give us, for upgrading deployments etc. so thanks again for sending this change over :)

Given it all works, I'm going to merge this down. Do feel free to send over any more enhancements if you make them in future; if one of us doesn't pick it up that quickly, tag me/someone else on the project, or assign it to one of us if you can (not sure on the repo permissions here. I'll check into it)

**Review by johadalin — APPROVED** [review]

**Comment by yujunz** [comment]
> assign it to one of us if you can (not sure on the repo permissions here. I'll check into it)

Thanks for the hint, @johadalin. It seems I don't have permission to make assignments.

**Comment by johadalin** [comment]
Cool. I'll see if it's possible to change that, and if not, just @ tag us :)

**Pull request merged**